classes/core/PKPQueueProvider.php

```diff
@@ -18,53 +18,162 @@
+            ->when(is_null($this->queue), fn($query) => $query->notQueue(PKPJobModel::TESTING_QUEUE))
+            ->notExceededAttempts();
+    }
+
+    /**
+     * Get the worker options object
+     */
+    public function getWorkerOptions(array $options = []): WorkerOptions
+    {
+        return new WorkerOptions(...WorkerConfiguration::withOptions($options)->getWorkerOptions());
+    }
+
+    /**
+     * Run the queue worker via an infinite loop daemon
+     */
+    public function runJobsViaDaemon(string $connection, string $queue, array $workerOptions = []): void
+    {
+        $laravelContainer = PKPContainer::getInstance();
+
+        $laravelContainer['queue.worker']->daemon(
+            $connection,
+            $queue,
+            $this->getWorkerOptions($workerOptions)
+        );
+    }
+    
+    /**
+     * Run the queue worker to process queue the jobs
+     */
+    public function runJobsInQueue(): void
+    {
+        $job = $this->getJobModelBuilder()->limit(1)->first();
 
         if ($job === null) {
             return;
         }
 
         $laravelContainer = PKPContainer::getInstance();
-        $options = new WorkerOptions(
-            'default',
-            $job->getDelay(),
-            $job->getAllowedMemory(),
-            $job->getTimeout(),
-            $job->getSleep(),
-            $job->getMaxAttempts(),
-            $job->getForceFlag(),
-            $job->getStopWhenEmptyFlag(),
-        );
 
         $laravelContainer['queue.worker']->runNextJob(
             'database',
             $job->queue,
-            $options
+            $this->getWorkerOptions()
         );
     }
+    
+    /**
+     * process the pending queue jobs at the end of current request/script end
+     */
+    public function runJobsAtShutdown(): void
+    {
+        $disableRun = Config::getVar('queues', 'disable_jobs_run_at_shutdown', false);
 
+        if ($disableRun || Application::isUnderMaintenance()) {
+            return;
+        }
+
+        $this->runJobsInQueue();
+    }
+
+    /**
+     * Bootstrap any application services.
+     *
+     * @return void
+     */
+    public function boot()
+    {
+        Queue::failing(function (JobFailed $event) {
+            app('queue.failer')->log(
+                $event->connectionName,
+                $event->job->getQueue(),
+                $event->job->getRawBody(),
+                $event->exception
+            );
+        });
+    }
+
+    /**
+     * Register the service provider.
+     *
+     * @return void
+     */
     public function register()
     {
-        register_shutdown_function([$this, 'runJobsAtShutdown']);
+        parent::register();
+
+        $this->registerDatabaseConnector(app()->get(\Illuminate\Queue\QueueManager::class));
+
+        $this->app->bind(
+            JobRepositoryInterface::class,
+            JobRepository::class
+        );
+
+        // register_shutdown_function([$this, 'runJobsAtShutdown']);
```

Comment:
May as well clear out the removed code!

Reply:
we can and will unless we decide to implement something similar proposed here https://github.com/pkp/pkp-lib/issues/8043#issuecomment-1214777409 .

Reply:
Done . Completely removed the functionality to remove the run jobs as through `register_shutdown_function`.